src/main/scala/com/timeout/kamon/cloudwatch/AmazonAsync.scala

```diff
@@ -0,0 +1,37 @@
+package com.timeout.kamon.cloudwatch
+
+import java.util.concurrent
+import java.util.concurrent.CancellationException
+
+import com.amazonaws.{AmazonWebServiceRequest, AmazonWebServiceResult, ResponseMetadata}
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.services.cloudwatch.AmazonCloudWatchAsync
+import com.amazonaws.services.cloudwatch.model.{MetricDatum, PutMetricDataRequest}
+import com.timeout.kamon.cloudwatch.KamonSettings.nameSpace
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{Future, Promise}
+import scala.util.Try
+
+object AmazonAsync {
+
+  type MetricDatumBatch = List[MetricDatum]
+
+  implicit class MetricsAsyncOps(data: MetricDatumBatch) {
+
+    def put(implicit client: AmazonCloudWatchAsync): Future[AmazonWebServiceResult[ResponseMetadata]] =
+      asyncRequest(new PutMetricDataRequest().withNamespace(nameSpace).withMetricData(data.asJava))(client.putMetricDataAsync)
+
+    private def asyncRequest[Arg, Req <: AmazonWebServiceRequest, Res](asyncArg: Arg)
```

Comment:
yes, this should be public and belong to AmazonAsync object which can be used by various types of async aws commands, i.e. MetricAsyncOps, S3AsyncOps, etc